### ISSUE TITLE
Custom interpolators sometimes do not compile

### ISSUE THREAD

**Issue by scf37** [lampepfl/dotty#16713]
## Compiler version
3.2.0

### Minimized code

```Scala
  q"$custom_checkbox $custom_control_input:indeterminate~$custom_control_label::before"
    .backgroundColor("#007bff")
```

### Output

```scala
[error] -- [E006] Not Found Error: /home/fine-css/src/test/scala/bootstrap/Bootstrap.scala:4427:21 
[error] 4427 |  q"$custom_checkbox $custom_control_input:indeterminate~$custom_control_label::before"
[error]      |                     ^^^^^^^^^^^^^^^^^^^^^
[error]      |                     Not found: $custom_control_input
[error]      |
[error]      | longer explanation available when compiling with `-explain`

```

But this compiles (with warning):
```Scala
  custom_control_input
  q"$custom_checkbox $custom_control_input:indeterminate~$custom_control_label::before"
    .backgroundColor("#007bff")
```

### Minimized code
```Scala
  q"$btn_outline_light:hover"
    .style("color", "#212529")
    .style("background-color", "#f8f9fa")
    .style("border-color", "#f8f9fa")
```

### Output
```Scala
[error] -- [E016] Syntax Error: /home/fine-css/src/test/scala/finecss/generated/TestModule2.scala:3513:22 
[error] 3513 |  q"$btn_outline_light:hover"
[error]      |                      ^
[error]      |          Error in interpolated string: identifier or block expected
[error]      |
[error]      | longer explanation available when compiling with `-explain`
[error] -- Error: /home/fine-css/src/test/scala/finecss/generated/TestModule2.scala:3514:4 
[error] 3514 |    .style("color", "#212529")
[error]      |    ^
[error]      |    end of statement expected but '.' found

```

But this compiles:
```Scala
  q"${btn_outline_light}:hover"
    .style("color", "#212529")
    .style("background-color", "#f8f9fa")
    .style("border-color", "#f8f9fa")
```

## Additional details

Compilation units are huge (about 10k lines) and contain hundreds of similar interpolations but only one or two of them breaks.
There are no errors on Scala 3.1.0

I was unable to make minimized example, they compile just fine
 Gists:
https://gist.github.com/scf37/7d3e0d551656399333e3e6ce36066fb4
https://gist.github.com/scf37/dd8aa8f113c1b4522c7847efadac76f4

On a side note, those compile like 1.5-2 times slower than Scala 2

**Comment by odersky** [comment]
We need more info and minimization to be able to diagnose this. What is `q`? I know its quasi-quotes in Scala 2, but in Scala 3 that has gone away. So what replaced it?

**Comment by scf37** [comment]
Here is full code if it helps: https://github.com/scf37/fine-css/blob/master/src/main/scala/finecss/Dsl.scala

Project compiles under 3.1.0 but `sbt Test/compile` will fail under 3.2.0 :+1: 

```
[error] -- [E016] Syntax Error: /home/asm/mywork/fine3/fine-css/src/test/scala/finecss/generated/TestModule2.scala:3513:22 
[error] 3513 |  q"$btn_outline_light:hover"
[error]      |                      ^
[error]      |          Error in interpolated string: identifier or block expected
[error]      |
[error]      | longer explanation available when compiling with `-explain`
[error] -- Error: /home/asm/mywork/fine3/fine-css/src/test/scala/finecss/generated/TestModule2.scala:3514:4 
[error] 3514 |    .style("color", "#212529")
[error]      |    ^
[error]      |    end of statement expected but '.' found
[error] -- [E006] Not Found Error: /home/asm/mywork/fine3/fine-css/src/test/scala/bootstrap/Bootstrap.scala:4427:21 
[error] 4427 |  q"$custom_checkbox $custom_control_input:indeterminate~$custom_control_label::before"
[error]      |                     ^^^^^^^^^^^^^^^^^^^^^
[error]      |                     Not found: $custom_control_input
[error]      |
[error]      | longer explanation available when compiling with `-explain`
[error] three errors found
[error] (Test / compileIncremental) Compilation failed
```

**Comment by odersky** [comment]
Yes, I guess we'd have to know what happens in `qSupport` then. The fact that it complains about 

    not found: $custom_control_input

is somewhat suspicious. It seems like it `qSupport` treats this as an identifier including the leading `$` whereas it probably should have stripped the `$`.

**Comment by som-snytt** [comment]
Project seems to compile under 3.2.1.

**Comment by scf37** [comment]
Yes, seems to be fixed in 3.2.1. Should've tested on 3.2.1 before opening this.
